### DISCUSSION TITLE
Autogeneration UTSettings.kt -> settings.properties (and probably some settings.md as well)

### DISCUSSION THREAD

**Discussion by Vassiliy-Kudryashov** [UnitTestBot/UTBotJava#1396] [Ideas]
{USER_HOME}/.utbot/settings.properties is configuration for **Intellij Plugin**, **CLI** and **CI** and obviously it should be well documented.
At the moment main "source of knowledge" is documentation inside the code, namely **UtSettings.kt**

The idea is to generate a template for **settings.properties** as resource file to be distributed together with the plugin from the existing sourcecode. 

'Template' here means that every _key=value_ line inside the file should be commented and relative documentation should be provided above it. One can easily uncomment the line and adjust the value to get desired behavior.

[Tomcat](https://github.com/apache/tomcat/blob/main/conf/catalina.properties) config file is well-known example of industry standard we may stick to.

**Comment by korifey** [comment]
I propose:

- Map all UI settings to this file (we intentionally sacrifice possibility to distinguish settings between projects)
- Settings from action dialog that "remembers user choice" (parametrized/source folders/framework selection) can be saved project-wide.

**Comment by Vassiliy-Kudryashov** [comment]
As for settings, the only "conflict-prone" intersection we have in fuzzing, namely global **useFuzzing=true/false** may conflict with local **fuzzingValue=0.0..1.0**
At the moment useFulling=false disables corresponding UI control at settings page and here [we have to show some hint with explanation](https://github.com/UnitTestBot/UTBotJava/issues/1411).

Except this unique case set of local UI settings don't insersect with global set of "core" or "low-level" settings.
So we shoudn't merge them and shouldn't sacrifice at all.

Plugin-specific settings
- mockStrategy
- staticsMocking
- runtimeExceptionTestsBehaviour
- hangingTestsTimeout
- forceStaticMocking
- treatOverflowAsError
- parametrizedTestSource
- classesToMockAlways
- commentStyle are not supported in CLI/CI and cannot be spread there quickly.

Instead we just should[ provide self-documented global config file (if absent) with all existing defaults for possible manual tuning](https://github.com/UnitTestBot/UTBotJava/issues/1412).
Note: we don't write into settings.properties, it's designed for reading only.
